doc/rtd/explanation/format.rst

```diff
@@ -24,93 +59,132 @@ These things may include:
 - importing certain SSH keys or host keys
 - *and many more...*
 
-See the :ref:`yaml_examples` section for a commented set of examples of
-supported cloud config formats.
+Many modules are available to process cloud-config data. These modules
+may run once per instance, every boot, or once ever. See the associated
+module to determine the run frequency.
 
-Begins with: ``#cloud-config`` or ``Content-Type: text/cloud-config`` when
-using a MIME archive.
-
-.. note::
-   Cloud config data can also render cloud instance metadata variables using
-   :ref:`jinja templates <instancedata-Using>`.
+See the :ref:`yaml_examples` section for a set of commented examples of
+supported cloud config formats.
 
 .. _user_data_script:
 
 User data script
 ================
 
-Typically used by those who just want to execute a shell script.
+Example
+-------
+
+.. code-block:: shell
+
+    #!/bin/sh
+    echo "Hello World" > /var/tmp/output.txt
 
-Begins with: ``#!`` or ``Content-Type: text/x-shellscript`` when using a MIME
-archive.
+Explanation
+-----------
 
-User data scripts can optionally render cloud instance metadata variables using
-:ref:`jinja templates <instancedata-Using>`.
+A user data script is a single shell script to be executed once per instance.
```

Comment:
It doesn't have to be a shell script does it? Wouldn't anything that can run with a shebang work (lua, perl, make, tcl, guile, etc)? Maybe just drop the word "shell"?

Reply:
> Wouldn't anything that can run with a shebang work 

Isn't that by definition a shell script then? Shell doesn't just mean `sh` or one of its descendants. Most of your examples there are listed as scripting languages [here](https://en.wikipedia.org/wiki/Scripting_language#Notable_languages)

Reply:
(ignore my other comment here if you saw it)